### ISSUE TITLE
Feature proposal: Liking posts / Post reactions

### ISSUE THREAD

**Issue by antoniobg** [thredded/thredded#499]
I was going to implement a "like this post" feature in a forum I'm building, but then I thought that it could be an interesting feature for the forum engine itself. I'm not sure if it's something the maintainers of the project would like to include, I would be happy to help if the answer is yes.

**Comment by glebm** [comment]
What do you think about making it a plugin instead? There are several ways to do the reactions themselves, and there are alternative approaches such as HN-style upvote/downvote, so it'd be nice if the choice was up to the user.

I think an architecture where the core is slim and provides extensions mechanisms for plugins is easier to to build on and maintain in the long run.

This seems like it would be a popular plugin, so we can install it with `thredded_create_app` by default, and use it in the demo app. We can also eventually host it on thredded.org.

**Comment by antoniobg** [comment]
Thanks for the answer. I agree that it makes sense to have it separately and do not bloat the core application with unnecessary features. I'll take a look at the plugin approach and will come back to you when I have some free time.

**Comment by glebm** [comment]
We don't have a good example plugin to base this on, but the UI can be extend by plugins via [view hooks](https://github.com/thredded/thredded#view-hooks) (you'd need to add some view hooks to the core for this which is perfectly fine).

Here is an example plugin: https://github.com/thredded/thredded-markdown_coderay.

**Comment by lukesarnacki** [comment]
@glebm I need that feature as well, but I am not sure I will have time to implement it as a plugin. I don't think that upvote / favourite buttons should be hidden on actions menu, so will you accept a PR with hook on post header? I would prefer it over copying header view.

**Comment by anushamummina** [comment]
@glebm Is the upvote/downvote feature included in threaded? I would like to use that feature in my forum. I am not getting any idea where and how exactly I need to use to achieve that feature. could you please guide me . 

Thanks in Advance

**Comment by glebm** [comment]
No, it is not included in Thredded and I don't know of any plugin that adds it.
